src/components/modals/CreateAntModal/CreateAntModal.tsx

```diff
@@ -0,0 +1,594 @@
+          ? 'pending'
+          : workflowStage + 1 > 2
+          ? 'success'
+          : '',
+    },
+    3: {
+      title: 'Deploy ANT',
+      status:
+        workflowStage + 1 === 3
+          ? 'pending'
+          : workflowStage + 1 > 3
+          ? 'success'
+          : '',
+    },
+    4: {
+      title: 'Manage ANT',
+      status: workflowStage + 1 === 4 ? 'success' : '',
+    },
+  };
+  const EDITABLE_FIELDS = [
+    'name',
+    'ticker',
+    'targetID',
+    'controller',
+    'owner',
+    'ttlSeconds',
+  ];
+
+  const ACTIONABLE_FIELDS: {
+    [x: string]: {
+      fn: () => void;
+      title: TRANSACTION_TYPES;
+    };
+  } = {
+    owner: {
+      fn: () => (ant.owner = JSON.stringify(modifiedValue)),
+      title: TRANSACTION_TYPES.ANT,
+    },
+  };
+
+  function reset() {
+    setWorkflowStage(0);
+    setAnt(new ANTContract());
+    setIsPostingTransaction(false);
+    setAntContractId(undefined);
+  }
+  // reset useEffect must be first, else wont reset
+  useEffect(() => {
+    if (!show) {
+      reset();
+    }
+    if (show) {
+      document.body.style.overflow = 'hidden';
+      return;
+    }
+    document.body.style.overflow = 'unset';
+  }, [show]);
+  useEffect(() => {
+    if (walletAddress) {
+      if (!ant.owner) {
+        ant.owner = walletAddress.toString();
+      }
+      if (!ant.controller) {
+        ant.controller = walletAddress.toString();
+      }
+      setDetails();
+    }
+  }, [walletAddress, ant, workflowStage]);
+
+  function setDetails() {
+    //  eslint-disable-next-line
+
+    const consolidatedDetails: any = {
+      name: ant.name,
+      ticker: ant.ticker,
+      targetID: ant.records['@'].transactionId,
+      ttlSeconds: ant.records['@'].ttlSeconds?.toString(),
+      controller: ant.controller,
+      owner: ant.owner,
+    };
+
+    const rows = Object.keys(consolidatedDetails).reduce(
+      (details: any, attribute: string, index: number) => {
+        const detail = {
+          attribute,
+          value: consolidatedDetails[attribute],
+          editable: EDITABLE_FIELDS.includes(attribute),
+          action: ACTIONABLE_FIELDS[attribute],
+          key: index,
+        };
+        details.push(detail);
+        return details;
+      },
+      [],
+    );
+    setRows(rows);
+  }
+
+  function handleStateChange() {
+    try {
+      if (!modifiedValue) {
+        throw new Error('Value is undefined');
+      }
+
+      if (editingField == 'name') {
+        ant.name = modifiedValue.toString();
+      }
+      if (editingField == 'ticker') {
+        ant.ticker = modifiedValue.toString();
+      }
+      if (editingField == 'owner') {
+        ant.owner = modifiedValue.toString();
+      }
+      if (editingField == 'controller') {
+        ant.controller = modifiedValue.toString();
+      }
+      if (editingField == 'targetID') {
+        ant.records = {
+          '@': {
+            transactionId: modifiedValue.toString(),
+          },
+        };
+      }
```

Comment:
nit: consider a switch statement instead of multiple `ifs`

Reply:
switched 😋